### ISSUE TITLE
Allow specifying separate http.Server and Express apps

### ISSUE THREAD

**Issue by christophemarois** [szymmis/vite-express#12]
Implementing a library like [socket.io](https://socket.io/) is difficult with current API. Socket.io's requires tying together an Express app and a Socket.io server through a common `node:http` server, e.g. (excerpt from their [Get Started](https://socket.io/get-started/chat) tutorial)

```typescript
const express = require('express');
const app = express();
const http = require('http');
const server = http.createServer(app);
const { Server } = require("socket.io");
const io = new Server(server);

app.get('/', (req, res) => {
  res.sendFile(__dirname + '/index.html');
});

io.on('connection', (socket) => {
  console.log('a user connected');
});

server.listen(3000, () => {
  console.log('listening on *:3000');
});
```

The quickest non-breaking way to make this possible with vite-express would be to either add a new export

```ts
// vite-express/src/main.ts
function bindToServer(
  httpServer: http.Server,
  expressApp: Express,
  port: number,
  callback?: () => void,
) {
  return httpServer.listen(port, async () => {
    await serveStatic(expressApp)
    await serveHTML(expressApp)
    if (Config.mode === 'development') await startDevServer()
    callback?.()
  })
}

// server/main.ts
ViteExpress.bindToServer(httpServer, expressApp, 3000, () =>
  console.log(
    `server listening at http://localhost:3000`,
  ),
)
```

Or to export `serveStatic`, `serveHTML`, `Config` and `startDevServer` and leave implementation details to the user.

What do you think?

**Comment by szymmis** [comment]
Hi @christophemarois, I believe that current API supports usage with socket.io as this problem was already mentioned in issue #3. You could add socket.io to the basic express app generated by `create-vite-express` and it would look something like this

```js
import express from "express";
import ViteExpress from "vite-express";
import { Server } from "socket.io";

const app = express();

app.get("/hello", (_, res) => {
  res.send("Hello Vite + React + TypeScript!");
});

const server = ViteExpress.listen(app, 3000, () =>
  console.log("Server is listening on port 3000...")
);

const io = new Server(server);

io.on("connection", (socket) => {
  console.log("User has conected!");

  socket.on("msg", (msg: string) => {
    console.log(`User sent a message: ${msg}`);
  });
});
```

I understand that this order might not be ideal and I support the idea of making the API as transparent as it can be, in order to be easily integrated in existing express apps without many changes, but we need to be careful in making such decisions to add additional functions etc. 

If this is not what you were thinking about, let me know, so we could decide on direction that we need to take in order to resolve it.

**Comment by fixmaker** [comment]
I too would be interested in something like this. I have a case where I would like a HTTPS version of my (vite-)express application.  Typically, this could be done with something like:
```
import express from 'express';
import https from 'https';

const app = express();

https.createServer(options, app).listen(443)
```
However, this doesn't seem possible with `vite-express` (unless I'm missing something) since `ViteExpress.listen(...)` calls `app.listen()` directly. @christophemarois suggestion would allow the HTTPS server to be injected into `vite-express`.

**Comment by szymmis** [comment]
That's a great example @fixmaker. I believe that you can use socket.io with vite-express in its current form, but what you showed might be indeed impossible to do right now. That's something that definitely needs to be done in order to make seamless integration possible. Thank you both for your contributions!

**Comment by szymmis** [comment]
Hi guys, this issue should be resolved by #19. I'll ship it in the next version. Let me know if that solves your use cases and what you think about this! Thanks a lot again

**Comment by eko2one** [comment]
@szymmis Could you please give an example on how to achieve https server with vite-express?
